### ISSUE TITLE
TF Models have no attribute .train() or .eval()

### ISSUE THREAD

**Issue by jamescolless** [huggingface/transformers#2617]
## 🐛 Bug

Using any of the TF models I am unable to set the **.eval()** or **.train()** properties. In addition, when loading from a pre-trained path (which the documentation seems to imply would mean that the models will be set to eval mode) I see non deterministic outputs given the same input indicating the models do not have dropout turned off.

Basic example:

```
model = TFBert.from_pretrained('path_to_model_directory', from_pt=True)
model.eval() #### This errors with 'TFBertModel# object has no attribute 'eval'
tokenizer = BertTokenizer.from_pretrained('path_to_model_directory')
inputs = tokenizer.encode_plus('Dummy text here.', return_tensors='tf')['input_ids']

print(model(inputs)) ## will have different outputs
print(model(inputs)) ## will not be the same
```

Any help would be greatly appreciated!

## Environment

* OS: Windows
* Python version: 3.6
* PyTorch version: 1.4
* PyTorch Transformers version (or branch): 2.3
* Using GPU ? Yes

**Comment by BramVanroy** [comment]
`model.eval()` is a PyTorch directive. It will disable dropout/norm, as you point out. On top of that, though, you'd also set the `no_grad` parameter so that weights are not updated.

Typically, your code'd look like this for inference/evaluation/testing.

```python
model.eval()
with torch.no_grad():
    # do stuff
```

I am not sure how this should be done in Tensorflow.

**Comment by sravanthi2019** [comment]
Is there a fix available for this because I am currently running a TFBertForSequenceClassification for classification and it gives non determinististic outcomes. I have written my code in tensorflow framework. Looks like the fix is placing model.eval() during evaluation operation as per https://github.com/google-research/bert/issues/583.
